src/Core/Utils/ObjectWithSemantic.hpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <cstdarg>
+#include <set>
+
+namespace Ra {
+namespace Core {
+namespace Utils {
+class RA_CORE_API ObjectWithSemantic
+{
+  public:
+    using SemanticName = std::string;
+
+    /// Store in set to allow for logarithmic search
+    using SemanticNameCollection = std::set<SemanticName>;
```

Comment:
not sure, according to the doc https://en.cppreference.com/w/cpp/container/set 
> Search, removal, and insertion operations have logarithmic complexity. Sets are usually implemented as red-black trees

So, the only expected improvement of using a sorted vector will be memory compacity. And as the proba of having high number of names is low, this will be negligible i think

Reply:
nevermind, set if ok for this usage.